filtering/macro.go

```diff
@@ -18,36 +21,47 @@ func ApplyMacros(filter Filter, declarations *Declarations, macros ...Macro) (Fi
 	return filter, nil
 }
 
-func applyMacros(exp *expr.Expr, sourceInfo *expr.SourceInfo, macros ...Macro) {
+func applyMacros(
+	exp *expr.Expr,
+	sourceInfo *expr.SourceInfo,
+	declarations *Declarations,
+	macros ...Macro,
+) []DeclarationOption {
+	declarationOptions := make([]DeclarationOption, 0, len(macros))
 	nextID := maxID(exp) + 1
 	Walk(func(currExpr, parentExpr *expr.Expr) bool {
 		cursor := &Cursor{
-			sourceInfo: sourceInfo,
-			currExpr:   currExpr,
-			parentExpr: parentExpr,
-			nextID:     nextID,
+			sourceInfo:       sourceInfo,
+			currExpr:         currExpr,
+			parentExpr:       parentExpr,
+			nextID:           nextID,
+			exprDeclarations: declarations,
 		}
```

Comment:
Could `declarations` come in here as `nil` and cause issues?
How about a nil check?

Reply:
It can indeed though unlikely. Nonetheless I added a check and setup an empty declaration if it is nil. See #362